crates/storage/provider/src/traits/executor.rs

```diff
@@ -1,25 +1,60 @@
 //! Executor Factory
 
-use crate::{post_state::PostState, StateProvider};
+use crate::{change::BundleStateWithReceipts, StateProvider};
 use reth_interfaces::executor::BlockExecutionError;
-use reth_primitives::{Address, Block, ChainSpec, U256};
+use reth_primitives::{Address, Block, BlockNumber, ChainSpec, PruneModes, U256};
+use std::time::Duration;
+use tracing::info;
 
 /// Executor factory that would create the EVM with particular state provider.
 ///
 /// It can be used to mock executor.
 pub trait ExecutorFactory: Send + Sync + 'static {
-    /// The executor produced by the factory
-    type Executor<T: StateProvider>: BlockExecutor<T>;
-
     /// Executor with [`StateProvider`]
-    fn with_sp<SP: StateProvider>(&self, sp: SP) -> Self::Executor<SP>;
+    fn with_state<'a, SP: StateProvider + 'a>(
+        &'a self,
+        _sp: SP,
+    ) -> Box<dyn PrunableBlockExecutor + 'a>;
 
     /// Return internal chainspec
     fn chain_spec(&self) -> &ChainSpec;
 }
 
+/// Block execution statistics
+#[derive(Clone, Debug, Default)]
+pub struct BlockExecutorStats {
+    /// How long execution took
+    pub execution_duration: Duration,
+    /// The amount of time it took to apply state changes to cached state
+    pub apply_state_duration: Duration,
+    /// Apply Post state execution changes duration.
```

Comment:
this doc is not very informative and uses outdated post state terminology

Reply:
It should have been `post execution state change` as the name implies we have state change post execution.
Would update docs